Feat vqa remote

MoHermes opened the pull request.

MoHermes committed:
feat: add VQA remote execution skeleton, to be modified

github-actions[bot] committed:
chore: Files formated

MoHermes committed:
feat: implement CMA-ES optimizer

MoHermes committed:
Merge branch 'feat-vqa-remote' of github.com:ColibrITD-SAS/mpqp into feat-vqa-remote

MoHermes committed:
feat: support qiskit runtime session in ibm_connection

MoHermes committed:
chore: bind parameters for qiskit and braket

MoHermes committed:
Merge remote-tracking branch 'origin/dev' into feat-vqa-remote

MoHermes committed:
Merge branch 'feat-to-other-language-AWS' into feat-vqa-remote

MoHermes committed:
Merge branch 'dev' into feat-vqa-remote

MoHermes committed:
feat: transpiled_circuit is device specific

MoHermes committed:
feat: support device specific pre-transpilation for observables

github-actions[bot] committed:
chore: Files formated

MoHermes committed:
fix: handle single atom case for Braket

MoHermes committed:
Merge branch 'feat-vqa-remote' of github.com:ColibrITD-SAS/mpqp into feat-vqa-remote

MoHermes committed:
chore: simplify handling Symbol

MoHermes committed:
fix: type check

MoHermes committed:
chore: avoid re-transpiling for the same device

MoHermes committed:
chore: modify device specific transpilation logic

MoHermes committed:
Merge branch 'dev' into feat-vqa-remote

MoHermes committed:
fix: type hinting

MoHermes committed:
Merge branch 'feat-to-other-language-AWS' of github.com:ColibrITD-SAS/mpqp into feat-vqa-remote

MoHermes committed:
chore: update session handling

MoHermes committed:
chore: enhance IBM session use

MoHermes committed:
feat: add unified execution pipeline, to support job, batch and session

MoHermes committed:
feat: support batch and session execution modes with IBM runtime

MoHermes committed:
feat: add VQAMode support to Job model

github-actions[bot] committed:
chore: Files formated

MoHermes committed:
chore: update bind_parameters in QCircuit

MoHermes committed:
chore: update optimizer behavior

MoHermes committed:
chore: update vqa implementation

MoHermes committed:
feat: handle bind parameters

MoHermes committed:
fix: improve session handling

github-actions[bot] committed:
chore: Files formated

MoHermes committed:
fix: avoid repeated observable pre_transpile conversion, update types

MoHermes committed:
chore: update job handling

MoHermes committed:
fix: bind parameters on transpiled circuit without mutating original

MoHermes committed:
chore: update remote execution flow

MoHermes committed:
chore: rename mode, simplify batch handling

MoHermes committed:
chore: update __init__ with ExecutionMode

MoHermes committed:
fix: resolve typing, apply_layout handling

MoHermes committed:
fix: remove invlaid pre_transpile, use translated_pre_measures

MoHermes committed:
fix: pre-transpilation caching per device

MoHermes committed:
fix: type annotations

MoHermes committed:
chore: update CMA-ES usage

MoHermes committed:
chore: update vqa, type issues need to be fixed

hJaffaliColibritd commented on mpqp/core/circuit.py:
```diff
@@ -2109,6 +2129,15 @@ def __repr__(self) -> str:
 
         return f'QCircuit({args_repr})'
 
+    def transpiled_for_device(self, device: AvailableDevice):
+        if device not in self.transpiled_circuit:
+            self.transpiled_circuit[device] = (
+                self.to_other_device(  # pyright: ignore[reportCallIssue]
+                    device  # pyright: ignore[reportArgumentType]
+                )
+            )
+        return self.transpiled_circuit[device]
+
```
For some providers, we would call to_other_device for each device , because MPQP translation and transpilation is device specific (like IBM), but for AWS/QLM for instance, we only need to_other_language (which is handled by to_other_device if I remember well), but then would need a different indexing of the dictionnary, otherwise we would export the mpqp circuit many times if we change the AWSDevice for instance (while we could have avoided).

hJaffaliColibritd committed:
Merge branch 'dev' of github.com:ColibrITD-SAS/mpqp into feat-vqa-remote

 Conflicts:
	mpqp/core/instruction/measurement/expectation_value.py
	mpqp/execution/providers/atos.py
	mpqp/execution/providers/aws.py
	mpqp/execution/providers/google.py
	mpqp/execution/providers/ibm.py
	mpqp/execution/runner.py
	mpqp/execution/vqa/vqa.py

github-actions[bot] committed:
chore: Files formated

JulienCalistoTD committed:
Merge branch 'dev' into feat-vqa-remote

github-actions[bot] committed:
chore: Files formated

JulienCalistoTD committed:
feat: Enhance VQA module with new optimizer structure and circuit handling

- Introduced VQAModule for managing multiple circuits and optimization processes.
- Updated optimizer functions to utilize OptimizerData for better parameter management.
- Refactored minimize function to streamline optimization logic and support new evaluation methods.
- Improved handling of ExpectationMeasure and observable pre-transpilation in QCircuit.
- Added tests for VQAModule across different providers to ensure functionality.
- Adjusted parameter binding logic in QCircuit to enhance flexibility.

JulienCalistoTD committed:
refactor: Remove unused import of minimize from VQA module

JulienCalistoTD committed:
Merge branch 'feat-vqa-remote' of https://github.com/ColibrITD-SAS/mpqp into feat-vqa-remote

github-actions[bot] committed:
chore: Files formated
